src/download.rs

```diff
@@ -68,35 +71,38 @@ pub fn download_file(url: &str, file_path: &Path, redirect_path: &Path) -> Resul
       }
       Ok(bytes_read) => {
         file.write_all(&buffer[..bytes_read])?;
-        downloaded += bytes_read as u64;
+        just_downloaded += bytes_read as u64;
+        let downloaded = file_size + just_downloaded;
 
         let elapsed = start.elapsed().as_secs_f64();
         let speed = if elapsed > 0.0 {
-          downloaded as f64 / elapsed
+          just_downloaded as f64 / elapsed
         } else {
           0.0
         };
         measurements.push_back(speed);
-        if measurements.len() > 10 {
+        if measurements.len() > MEASUREMENT_SIZE {
           measurements.pop_front();
         }
         let avg_speed = measurements.iter().sum::<f64>() / measurements.len() as f64;
-        let eta = if avg_speed > 0.0 {
-          (total_size as f64 - downloaded as f64) / avg_speed
+        let eta = if avg_speed > 1.0 && measurements.len() > (MEASUREMENT_SIZE / 2) {
+          Eta::Seconds((total_size as f64 - downloaded as f64) / avg_speed)
```

Comment:
Now that I think of it, it would probably be nice to extend the `Eta` and put all calculations there so it's responsible for _estimating_ ;) It would unclutter the code.

Reply:
Maybe next time :)